lib/matplotlib/dates.py

```diff
@@ -405,18 +405,20 @@ def date2num(d):
     Gregorian calendar is assumed; this is not universal practice.
     For details see the module docstring.
     """
-
     if hasattr(d, "values"):
         # this unpacks pandas series or dataframes...
         d = d.values
-
-    if ((isinstance(d, np.ndarray) and np.issubdtype(d.dtype, np.datetime64))
-            or isinstance(d, np.datetime64)):
-        return _dt64_to_ordinalf(d)
     if not np.iterable(d):
+        if (isinstance(d, np.datetime64) or (isinstance(d, np.ndarray) and
```

Comment:
I don't think that we will ever hit the second half of this conditional as 

```python

In [5]: np.iterable(np.arange(5))
Out[5]: True

```

Reply:
```python
t0 = datetime.datetime(2017, 1, 1, 0, 1, 1)
tnp = np.array(t0, dtype='datetime64[us]')
print(np.iterable(tnp))
```
returns `False` (believe it or not)...